cli/cli/commands/run/run.go

```diff
@@ -343,24 +343,18 @@ func readAndPrintResponseLinesUntilClosed(responseLineChan <-chan *kurtosis_core
 	}
 	defer printer.Stop()
 
-	isRunSuccessful := false // defaults to false such that we fail loudly if something unexpected happens
 	for {
 		select {
 		case responseLine, isChanOpen := <-responseLineChan:
+			// we do not want to show go errors at the end in case of APIC failures
```

Comment:
I suggest do not remove the functionality in the backend, I will prefer removing directly in the CLI, in the `run` command:
`errRunningKurtosis = readAndPrintResponseLinesUntilClosed(responseLineChan, cancelFunc, verbosity, dryRun)
	if errRunningKurtosis != nil {
		servicesInEnclaveForMetrics, servicesInEnclaveForMetricsError := enclaveCtx.GetServices()
		if servicesInEnclaveForMetricsError != nil {
			logrus.Warn("Tried getting number of services in the enclave to log metrics but failed")
		} else {
			if err := metricsClient.TrackKurtosisRunFinishedEvent(starlarkScriptOrPackagePath, len(servicesInEnclaveForMetrics), runFailed); err != nil {
				logrus.Warn("An error occurred tracking kurtosis run finished event")
			}
		}

		return nil
	}`

doing it this way we won't lose the `metrics reporting part`.